controllers/suite_test.go

```diff
@@ -82,8 +126,13 @@ var _ = BeforeSuite(func() {
 	})
 	Expect(err).ToNot(HaveOccurred())
 
+	k8sClient = k8sManager.GetClient()
+	Expect(k8sClient).ToNot(BeNil())
+
+	cclient = CustomClient{k8sClient, nil}
```

Comment:
Nit: IMO it's a bit messy to initialize it like it especially because `onDeleteError` is never relevant for the initialization.
I suggest either:
set up a constructor that only takes the client, 
or something like this:
```
cclient = CustomClient{Client: k8sClient}
```